app/products/calls/websocket.ts

```diff
@@ -86,12 +127,35 @@ export default class WebSocketClient extends EventEmitter {
 
     close() {
         if (this.ws) {
+            this.closed = true;
             this.ws.close();
             this.ws = null;
+            this.seqNo = 1;
+            this.serverSeqNo = 0;
+            this.connID = '';
+            this.originalConnID = '';
+        } else {
```

Comment:
For my own understanding, under which use case would `this.ws` be null to have the `else block` triggered?

Reply:
That would be any time an unexpected disconnection happened as we are setting `this.ws = null` in the `onclose` handler before calling `close()`. The other case instead is when a client explicitly wants to disconnect (e.g. leaving a call).

Reply:
That sounds correct to me. That's what I meant by "the other case", the one in which `this.ws != null`. Am I missing something? :fearful:

Reply:
Ah! That's what you meant by "the other case" -- hah. I thought you meant "another way it could be null". 😅